### PR TITLE
[PE-1652] add core tests

### PR DESCRIPTION
**Pull request by prateekdefi**
Issue: https://linear.app/perennial/issue/PE-1652/[testing]-increase-core-test-coverage
Add unit and integration tests to increase core coverage

**Unit test coverage before:**
```
-----------------------------|----------|----------|----------|
File                         | % Branch |  % Funcs |  % Lines |
-----------------------------|----------|----------|----------|
 contracts/                  |    85.38 |    89.06 |    95.28 |
  Market.sol                 |       83 |    86.67 |    94.84 |
  MarketFactory.sol          |    93.33 |    94.74 |    97.56 |
 contracts/types/            |    95.77 |    95.88 |     96.7 |
  Guarantee.sol              |      100 |    90.91 |    96.67 |
  Order.sol                  |    94.83 |    96.77 |    98.73 |
  Position.sol               |    84.62 |    93.55 |    85.96 |
-----------------------------|----------|----------|----------|
```

**Unit test coverage after:**
```
-----------------------------|----------|----------|----------|
File                         | % Branch |  % Funcs |  % Lines |
-----------------------------|----------|----------|----------|
 contracts/                  |    98.46 |    98.44 |    99.61 |
  Market.sol                 |       98 |    97.78 |    99.53 |
  MarketFactory.sol          |      100 |      100 |      100 |
 contracts/types/            |    95.77 |    98.97 |    97.69 |
  Guarantee.sol              |      100 |      100 |      100 |
  Order.sol                  |    94.83 |      100 |      100 |
  Position.sol               |    84.62 |    96.77 |    87.72 |
-----------------------------|----------|----------|----------|
```

**Integration test coverage before:**
```
-----------------------------|----------|----------|----------|
File                         | % Branch |  % Funcs |  % Lines |
-----------------------------|----------|----------|----------|
 contracts/                  |       50 |    67.19 |    75.59 |
  Market.sol                 |       53 |    77.78 |    82.16 |
  MarketFactory.sol          |       40 |    42.11 |    41.46 |
 contracts/libs/             |       65 |      100 |    96.72 |
  CheckpointLib.sol          |       90 |      100 |      100 |
  VersionLib.sol             |     87.5 |      100 |    99.18 |
 contracts/types/            |    51.15 |    91.75 |    90.43 |
  Guarantee.sol              |    38.89 |    81.82 |       80 |
  Order.sol                  |    55.17 |    93.55 |    87.34 |
-----------------------------|----------|----------|----------|
```

**Integration test coverage after:**
```
-----------------------------|----------|----------|----------|
File                         | % Branch |  % Funcs |  % Lines |
-----------------------------|----------|----------|----------|
 contracts/                  |    93.08 |    98.44 |    99.61 |
  Market.sol                 |       92 |    97.78 |    99.53 |
  MarketFactory.sol          |    96.67 |      100 |      100 |
 contracts/libs/             |    68.75 |      100 |    96.72 |
  CheckpointLib.sol          |      100 |      100 |      100 |
  VersionLib.sol             |    95.83 |      100 |    99.18 |
 contracts/types/            |    54.62 |    93.81 |    94.39 |
  Guarantee.sol              |    61.11 |    90.91 |    96.67 |
  Order.sol                  |    63.79 |    96.77 |     96.2 |
-----------------------------|----------|----------|----------|
```

### CODE REVIEW

File: packages/perennial/test/unit/market/Market.test.ts [hunk 1]
```diff
@@ -24044,5 +24071,30 @@ describe('Market', () => {
         )
       })
     })
+
+    describe('#claimExpoxure', async () => {
```
[the remark this answers is not in the document]
Comment (prateekdefi) [reply packages/perennial/test/unit/market/Market.test.ts]: Done [ba77da9](https://github.com/equilibria-xyz/perennial-v2/pull/428/commits/ba77da92577fd2b2a0945160ef4346374f4d8421)

File: packages/perennial/test/unit/market/Market.test.ts [hunk 2]
```diff
@@ -24052,5 +24116,174 @@ describe('Market', () => {
...
+        await market.connect(owner).updateParameter(await market.parameter())
+        await expect(market.connect(coordinator).claimExposure()).to.be.revertedWithCustomError(
+          market,
+          'InstanceNotOwnerError',
+        )
+      })
+    })
+  })
+
+  describe('reentrancy', async () => {
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:24236]: Again, this seems more appropriate to include with PE-87.

**Review by EdNoepel — COMMENTED** [review]
Recommend moving the reentrancy tests to a different branch, where they could possibly be added alongside oracle price verification reentrancy checks.

File: packages/perennial/test/unit/market/Market.test.ts (hunk 2 above)
Comment (EdNoepel) [reply packages/perennial/test/unit/market/Market.test.ts:24236]: Per discussion, leaving here as this test is needed for full core coverage.

File: packages/perennial/contracts/test/MockToken.sol
[the remark this answers is not in the document]
Comment (prateekdefi) [reply packages/perennial/contracts/test/MockToken.sol:1]: [PE-87](https://linear.app/perennial/issue/PE-87/add-re-entrancy-integration-tests-to-perennial) mentions of re-entrancy integration tests while I have added unit tests here. So I think we can keep these reentrancy tests in the same PR as this is required to increase coverage to 100% for core.

**Comment by prateekdefi** [comment]
> Recommend moving the reentrancy tests to a different branch, where they could possibly be added alongside oracle price verification reentrancy checks.

[PE-87](https://linear.app/perennial/issue/PE-87/add-re-entrancy-integration-tests-to-perennial) mentions of re-entrancy integration tests while I have added unit tests here. So I think we can keep these reentrancy tests in the same PR as this is required to increase coverage to 100% for core.

**Review by EdNoepel — APPROVED** [review]
Core unit test coverage from 97.03% to 98.92%.
Core integration test coverage went from 86.89% to 93.92%.

**Review by EdNoepel — APPROVED** [review]
Would be nice to also test that `factory.updateExtension` and `factory.updateOperator` behave as expected when passed `enabled=false`.  But this is improvement.

**Review by kbrizzle — APPROVED** [review]

**Pull request merged**